src/dune_sexp/syntax.ml

```diff
@@ -270,22 +333,27 @@ let create ?(experimental = false) ~name ~desc supported_versions =
 let name t = t.name
 
 let check_supported ~dune_lang_ver t (loc, ver) =
-  if
-    not (Supported_versions.is_supported t.supported_versions ver dune_lang_ver)
-  then
+  match Supported_versions.status t.supported_versions ver ~dune_lang_ver with
+  | `Supported -> ()
+  | `Deleted_in deleted_in ->
+    User_error.raise ~loc
+      [ Pp.textf
+          "Version %s of the %s extension has been deleted in Dune %s. Please \
```

Comment:
might be useful to mention what's the newest (or oldest) version available in this mesage

Reply:
You can directly add a hint with the current dune lang version.

Reply:
Done!